docs/guides/color.md

```diff
@@ -0,0 +1,320 @@
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1],
+])
+points.face_color = np.array([
+    [1, 0, 0, 1],
+    [0, 1, 0, 1],
+    [0, 0, 1, 1],
+])
+```
+
+However, when specifying many colors in a sequence or generator *do not mix different single color types*.
+For example, something like
+
+```python
+In [5]: points.face_color = np.array(['red', (0, 1, 0), '#0000ff'])
+```
+
+is not supported or guaranteed to work.
+
+
+## Colormaps
+
+In napari, a colormap maps data values to RGBA colors and is used in at least two ways.
+
+- To map an image pixel value to an RGBA color.
+- To map a property value to an RGBA color.
+
+In both cases, the types of data values may either be continuous (e.g. `float`) or categorical (e.g. `int`, `str`),
+so napari has two different types of colormaps to handle those two cases.
+
+
+### Continuous maps
+
+The [`Colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap.py#L28)
+class defines a map from a continuous or real value to a color.
+For example, a 2D image of floats between 0 and 1 can be mapped to colors as follows.
+
+```python
+In [6]: image.colormap = 'blue'
+```
+
+Internally, the [`ensure_colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap_utils.py#L492)
+function is used to coerce colormap input values to an instance of napari's [`Colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap.py#L28).
+
+```python
+In [7]: image.colormap
+Out[7]:
+Colormap(
+    colors=np.array([0., 0., 0., 1], [0., 0., 1., 1]]),
+    name='blue',
+    interpolation=<ColormapInterpolationMode.LINEAR: 'linear'>,
+    controls=array([0., 1.]),
+)
+```
+
+In the `Colormap` object returned, the `controls` attribute defines the
+input data values that the colormap maps from and the `colors` attribute
+defines the output colors that it maps to.
+The `interpolation` attribute defines how to interpolate `color` when an
+input value does not exactly equal one of the values in `controls`.
+And the `name` attribute gives the colormap a name that can be referred to later.
+
+In this case, input data values of 0 are mapped to black (`[0, 0, 0, 1]`),
+values of 1 are mapped to blue (`[0, 0, 1, 1]`),
+and values in `(0, 1)` are mapped to colors that are linearly interpolated
+between black and blue.
+
+#### Supported input
+
+A few types of input are supported.
+The easiest to use is the colormap name as above.
+
+```python
+image.colormap = 'blue'
+```
+
+By default colormap names come from one of the following sources.
+
+- Some of [matplotlib's colormap names](https://matplotlib.org/stable/tutorials/colors/colormaps.html).
+- Standard optical primary and secondary colors: `'red', 'green', 'blue', 'yellow', 'magenta', 'cyan'`.
+- BOP (blue, orange, purple) colormap names: `'bop blue', 'bop orange', 'bop purple'`.
```

Comment:
I'm guessing there is some implicit knowledge here, I don't know what/why BOP are a thing I found https://github.com/cleterrier/ChrisLUTs, so maybe they come from ImageJ, and they look like complementary, but maybe add a reference ?

Reply:
Yep, good idea. I remember when I search for bop colormap, I more easily found a minecraft mod call biomes o' plenty, so having an explicit reference is a good idea!

I also assume that repo you reference is the source for these colormaps, so I added that for now. But if anyone knows otherwise, please correct that!